src/query/storages/system/src/tables_table.rs

```diff
@@ -312,19 +310,19 @@ where TablesTable<T, U>: HistoryAware
 
                 for table in tables {
                     let table_id = table.get_id();
-                    // If db1 is visible, do not means db1.table1 is visible. An user may have a grant about db1.table2, so db1 is visible
+                    // If db1 is visible, do not mean db1.table1 is visible. A user may have a grant about db1.table2, so db1 is visible
                     // for her, but db1.table1 may be not visible. So we need an extra check about table here after db visibility check.
                     if visibility_checker.check_table_visibility(
                         ctl_name,
-                        db.name(),
+                        db_name,
                         table.name(),
                         db_id,
                         table_id,
                     ) && table.engine() != "STREAM"
                     {
                         if !U && table.get_table_info().engine() == "VIEW" {
-                            catalogs.push(ctl_name);
-                            databases.push(name);
+                            catalogs.push(ctl_name.as_str());
+                            databases.push(db_name.to_owned());
```

Comment:
The above ctl_name.`as_str` but db_name.`to_owned`?

Reply:
lifetime issues:

 `ctrl_name`'s type is &String, and has a longer enough lifetime to be pushed into `catalogs` as `&str`
 
 `db_name`'s lifetime is shorter, unfortunately, therefore, it need to be "converted" to a `String`, and pushed (and move the ownership) into `databases`